app/controllers/application_controller.rb

```diff
@@ -1,3 +1,16 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  add_flash_types :success, :danger, :warning
+
+  include SessionsHelper
+
+  private
+
+  def logged_in_user
+    return if logged_in?
+    store_location
+    flash[:danger] = "Please log in."
+    redirect_to login_url
```

Comment:
`redirect_to login_url, danger : "Please log in."` thì có được không?

Reply:
dạ không ạ!! em đã fix